src/include/firebird/FirebirdInterface.idl

```diff
@@ -1487,3 +1507,51 @@ interface DecFloat34 : Versioned
 	void fromBcd(int sign, const uchar* bcd, int exp, FB_DEC34* to);
 	void fromString(Status status, const string from, FB_DEC34* to);
 }
+
+// Replication interfaces
+
+interface ReplicatedRecord : Versioned
```

Comment:
I see no method to get record's format or any other way to get separate fields values. Does it exist?

Reply:
It doesn't exist yet, but I plan to add it after Beta. Your suggestions are appreciated.

Reply:
Avalerion uses this:
```
class IField
{
public:
	virtual bool hasData() = 0;
	virtual const char* getName() = 0;
	virtual int getType() = 0;
	virtual int getScale() = 0;
	virtual int getLength() = 0;
	virtual int getCharset() = 0;
	virtual bool isNull() = 0;
	virtual void* getData() = 0;
	virtual bool next() = 0;
};

class IRecord
{
public:
	virtual const char* getSchemaName() = 0;
	virtual const char* getTableName() = 0;
	virtual long long getNumber() = 0;
	virtual IField* getField() = 0;
};
```

Where IField is actually enumerator. I would like to have also "bool isKey()" in it.